pricelist_cache/models/product_pricelist_cache.py

```diff
@@ -150,6 +150,13 @@ def _get_product_ids_to_update(self, pricelist, product_ids):
                     lambda i: i.product_id.id in product_ids
                 )
                 product_ids_to_update = product_item_ids.mapped("product_id").ids
+                template_items = pricelist.item_ids.filtered(
+                    lambda i: i.product_tmpl_id.product_variant_ids
+                )
+                if template_items:
+                    variants_ids = template_items.product_tmpl_id.product_variant_ids
+                    if any([True for id in variants_ids.ids if id in product_ids]):
+                        product_ids_to_update.extend(variants_ids.ids)
```

Comment:
This would be good to have it in one loop with the first. item_ids.filtered

Reply:
updated